cmake/podioMacros.cmake

```diff
@@ -151,3 +151,40 @@ function(PODIO_GENERATE_DATAMODEL datamodel YAML_FILE RETURN_HEADERS RETURN_SOUR
     )
 
 endfunction()
+
+#---------------------------------------------------------------------------------------------------
+function(PODIO_CHECK_CPP_FS FS_LIBS)
+  SET(${have_filesystem} FALSE)
+  MESSAGE(VERBOSE "Checking for filesystem support of compiler")
+  # GNU implementation prior to 9.1 requires linking with -lstdc++fs and LLVM
+  # implementation prior to LLVM 9.0 requires linking with -lc++fs
+  # After that it should be built-in
+  FOREACH(FS_LIB_NAME "" stdc++fs c++fs)
+    MESSAGE(VERBOSE "Linking against ${FS_LIB_NAME}")
+    try_compile(have_filesystem ${CMAKE_BINARY_DIR}/try ${PROJECT_SOURCE_DIR}/cmake/try_filesystem.cpp
+      CXX_STANDARD ${CMAKE_CXX_STANDARD}
+      CXX_EXTENSIONS False
+      OUTPUT_VARIABLE HAVE_FS_OUTPUT
+      LINK_LIBRARIES ${FS_LIB_NAME}
+      )
+    MESSAGE(DEBUG "-----> " ${HAVE_FS_OUTPUT})
+    IF(have_filesystem)
+      MESSAGE(VERBOSE "Compiler supports filesystem when linking against ${FS_LIB_NAME}")
+      SET(${FS_LIBS} ${FS_LIB_NAME} PARENT_SCOPE)
+      BREAK()
+    ENDIF()
+    MESSAGE("Compiler not compatible when linking against ${FS_LIB_NAME}")
+  ENDFOREACH()
+
+  IF(NOT have_filesystem)
+    MESSAGE(STATUS "Compiler does not have filesystem support, falling back to boost::filesystem")
+    find_package(Boost REQUIRED COMPONENTS filesystem system)
+    SET(${FS_LIBS} Boost::filesystem Boost::system PARENT_SCOPE)
+    SET_TARGET_PROPERTIES(Boost::filesystem
+      PROPERTIES
+      INTERFACE_COMPILE_DEFINITIONS USE_BOOST_FILESYSTEM
+    )
+  GET_TARGET_PROPERTY(BOOST_FILESYSTEM_LOC Boost::filesystem IMPORTED_LOCATION)
```

Comment:
These two lines don't do anything in this case. I am not even sure of they do anything in the DD4hep macro case.

Reply:
Ah, I see. I didn't realize that. I have removed them.